bin/amazon_ssa_agent

```diff
@@ -107,35 +57,17 @@ begin
   aws_args[:s3]  = Aws::S3::Resource.new(region: reg)
 
   #
-  # Logging args.
+  # Initialize Logging
   #
-  aws_args[:log_level] = log_level_str unless log_level_str.nil? # command line override
-  log_level            = LOG_LEVELS[aws_args[:log_level]]
-  max_log_size       ||= aws_args[:max_log_size] || 1024 * 256
-  log_file             = File.join('/opt/miq/log', 'extract.log')
-  log_args = {
-    :formatter => LogFormatter,
-    :filename  => log_file,
-    :maxsize   => max_log_size,
-    :aws_args  => aws_args
-  }
-
-  #
-  # Initialize logging.
-  #
-  LogFormatter.extractor_id = extractor_id
-  $log = Log4r::Logger.new('toplog')
-  $log.level = log_level
-  lfo = Log4r::RollingS3Outputter.new('log_s3', log_args)
-  lfo.only_at(Log4r::DEBUG, Log4r::INFO, Log4r::WARN, Log4r::ERROR, Log4r::FATAL, Log4r::COPY)
-  $log.add('log_s3')
-  at_exit { lfo.flush }
-
-  if log_to_stderr
-    eso = Log4r::StderrOutputter.new('err_console', :formatter=>LogFormatter)
-    eso.only_at(Log4r::DEBUG, Log4r::INFO, Log4r::WARN, Log4r::ERROR, Log4r::FATAL, Log4r::COPY)
-    $log.add('err_console')
-  end
+  $log = AmazonSsaSupport::RollingS3Logger.new(
+    File.join('/opt/miq/log', 'extract.log'),
+    :maxsize       => max_log_size || aws_args[:max_log_size] || 1024 * 256,
+    :aws_args      => aws_args,
+    :log_to_stderr => log_to_stderr,
+    :level         => Logger.const_get((log_level_str || aws_args[:log_level] || "INFO").upcase),
+    :progname      => File.basename(__FILE__, ".*"),
```

Comment:
level and progname are already supported parameters of Logger, so this uses those directly.